server/src/main/java/io/crate/execution/ddl/tables/MappingUtil.java

```diff
@@ -185,11 +202,20 @@ private static Map<String, Map<String, Object>> createPropertiesMap(AllocPositio
         }
         HashMap<String, Map<String, Object>> allColumnsMap = new LinkedHashMap<>();
         for (Reference child: children) {
-            allColumnsMap.put(child.column().leafName(), addColumnProperties(position, child, tree));
+            allColumnsMap.put(mappingKey(child), addColumnProperties(position, child, tree));
         }
         return allColumnsMap;
     }
 
+    private static String mappingKey(Reference reference) {
+        if (reference.isDropped()) {
+            assert reference.oid() != COLUMN_OID_UNASSIGNED : "Only columns with assigned OID-s can be dropped";
+            return String.format(Locale.ENGLISH, "%s%d........", DROPPED_COLUMN_NAME_PREFIX, reference.oid());
+        } else {
```

Comment:
In Postgres it's column ordinal as they don't have column OID. 

I used OID in case want to look into source and check what values had the specific dropped column. Should I also include human readable name here ? Otherwise we lose it since it's taken fro the mapping key currently which is now prefix + OID.